x-pack/plugins/actions/server/builtin_action_types/lib/send_email.ts

```diff
@@ -50,12 +62,92 @@ export interface Content {
 // send an email
 export async function sendEmail(logger: Logger, options: SendEmailOptions): Promise<unknown> {
   const { transport, routing, content, configurationUtilities, hasAuth } = options;
-  const { service, host, port, secure, user, password } = transport;
+  const { service, clientId, clientSecret, tenantId, oauthTokenUrl } = transport;
   const { from, to, cc, bcc } = routing;
   const { subject, message } = content;
 
+  const messageHTML = htmlFromMarkdown(logger, message);
+
+  if (service === AdditionalEmailServices.EXCHANGE) {
```

Comment:
This function is getting very confusing 😬. Could we at least pull this if block out into another function? If we want to clean it up further, maybe something like this:

```
export async function sendEmail(logger: Logger, options: SendEmailOptions): Promise<unknown> {
  const messageHTML = htmlFromMarkdown(logger, message);

  if (service === AdditionalEmailServices.EXCHANGE) {
    await sendEmailWithExchange(messageHTML);
  } else {
    await sendEmailWithNodemailer(messageHTML);
  }
}
```

and then encapsulate using those functions? Just a suggestion...feel free to ignore :). I just had a hard time following all the logic in this function